crates/http/src/routes/ws.rs

```diff
@@ -0,0 +1,10 @@
+use axum::Json;
+use serde_json::Value;
+
+use crate::Result;
+
+pub(crate) async fn get_peers_by_domain_handler() -> Result<Json<Value>> {
+    Ok(Json(serde_json::to_value(
+        &(iron_ws::commands::ws_peers_by_domain().await),
+    )?))
```

Comment:
why change the return value here instead of keeping the original `Vec<Peer>`? I would assume it serializes automatically

Reply:
Couple of reasons. First, the `peers` module in `crates/ws` is private, so we can't use the `Peer` struct without making it public to other crates. Not an issue, just a statement.

The other reason is that `HashMap` does not implement the `Handler` trait that Axum requires for functions that handle requests. It seemed simpler convert to JSON than to deal with implementing the trait

Doing this
```rust
pub(crate) async fn get_peers_by_domain_handler() -> Result<HashMap<String, Vec<Peer>>> {
    Ok(iron_ws::commands::ws_peers_by_domain().await)
}
```

results in the following error

```
the trait bound `fn() -> impl std::future::Future<Output = std::result::Result<std::collections::HashMap<std::string::String, std::vec::Vec<iron_ws::peers::Peer>>, error
::Error>> {routes::ws::get_peers_by_domain_handler}: axum::handler::Handler<_, _, _>` is not satisfied
the following other types implement trait `axum::handler::Handler<T, S, B>`:
  <axum::handler::Layered<L, H, T, S, B, B2> as axum::handler::Handler<T, S, B2>>
  <axum::routing::MethodRouter<S, B> as axum::handler::Handler<(), S, B>> [E0277]
```

Is this what you meant or did you had something else in mind?

Reply:
You're missing the `Json(_)` part in that last example which I think is the reason why it fails to compile

Reply:
You're quite right, adding `Json(_)` solves the issue. I'm going to update this to keep the same return value